utb/views/submit_report.py

```diff
@@ -25,14 +40,75 @@ def handler(request):
         return HttpResponse("Article not found", status=404)
     article = qs[0]
 
+    previous_status = article.get_status()
+
     report = Report(user=user,
                     article=article,
-                    report_value=report)
+                    value=report_value.name)
     report.save()
-    if report == "L":
-        article.positive_reports += 1 * user.multiplier
-    else:
-        article.negative_reports += 1 * user.multiplier
+
     user.n_reports += 1
-    article.save()
     user.save()
+
+    if report_value == Report.Values.L:
+        article.legit_reports += user.weight
+    else:
+        article.fake_reports += user.weight
+
+    updated_status = article.get_status()
+    if previous_status != updated_status:
+        change_article_status(report, article, previous_status, updated_status)
+    return HttpResponse("Created", status=201)
+
+
+def change_article_status(last_report, article, previous_status, updated_status):
```

Comment:
change article status is pretty complex, do you mind adding some comments to make the code clearer?